lib/pure/streams.nim

```diff
@@ -1274,7 +1274,7 @@ else: # after 1.3 or JS not defined
 
     new(result)
     result.data = s
-    when defined(gcOrc) or defined(gcArc):
+    when not defined(js) and not defined(nimscript) and not defined(nimSeqsV2):
```

Comment:
`not defined(nimSeqsV2)`? Why not

```suggestion
    when (defined(gcOrc) or defined(gcArc)) and not defined(js) and not defined(nimscript):
```

Reply:
It's what `preparemutations` is [guarded with](https://github.com/nim-lang/Nim/blob/version-1-6/lib/system.nim#L3141).
So figured that it was the best.

Reply:
In hindsight `when declared(prepareMutation)` seems more 'proper'